staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -347,3 +349,38 @@ func GetHistogramMetricValue(m metrics.ObserverMetric) (float64, error) {
 	}
 	return metricProto.Histogram.GetSampleSum(), nil
 }
+
+func LabelsMatch(metric *dto.Metric, labelFilter map[string]string) bool {
+	for _, lp := range metric.GetLabel() {
+		if value, ok := labelFilter[lp.GetName()]; ok && lp.GetValue() != value {
+			return false
+		}
+	}
+	return true
+}
+
+// ExpectFindMetric find a metric with the given name nad labels or reports a fatal test error.
+func ExpectFindMetric(t *testing.T, name string, expectedLabels map[string]string) *dto.Metric {
```

Comment:
I'm afraid we can't just simply move `LabelsMatch` and `ExpectFindMetric` to components.

There are a few methods you can select in testutil package, such as `CollectAndCompare` and `GatherAndCompare`, etc.
That means you need to refactor the original test case with the methods in testutil.

Reply:
sure , thanks your comment , i will udpate it